hack/lib/golang.sh

```diff
@@ -158,11 +163,20 @@ readonly KUBE_TEST_PORTABLE=(
 # need to target server platforms.
 # These binaries will be distributed in the kubernetes-test tarball.
 # If you update this list, please also update build/release-tars/BUILD.
-readonly KUBE_TEST_SERVER_TARGETS=(
-  cmd/kubemark
-  vendor/github.com/onsi/ginkgo/ginkgo
-  test/e2e_node/e2e_node.test
-)
+kube::golang::server_test_targets() {
+  local targets=(
+    cmd/kubemark
+    vendor/github.com/onsi/ginkgo/ginkgo
+  )
+
+  if [[ "${OSTYPE:-}" == "linux"* ]]; then
```

Comment:
hm, interesting, I suspect this still only works by accident - the `$OSTYPE` in your docker container is probably set to linux.

This previously wasn't an issue because we only cross-build `KUBE_TEST_SERVER_TARGETS` for linux platforms.
I think you can only run into the compilation failure if you try to build everything (i.e. just run `make`) on macOS natively, without docker or anything else.

If you tried cross-building on macOS natively I think this will do the wrong thing. But I'm not sure this is a use case we need to worry about.

Further thoughts: we should perhaps not include `KUBE_SERVER_TARGETS` or `KUBE_TEST_SERVER_TARGETS` in `KUBE_ALL_TARGETS` when building on non-linux, since neither of those sets is expected to be built for non-linux.